src/v/storage/log_reader.cc

```diff
@@ -268,6 +269,17 @@ log_reader::do_load_slice(model::timeout_clock::time_point timeout) {
         return _iterator.close().then(
           [] { return ss::make_ready_future<storage_t>(); });
     }
+    /**
```

Comment:
something this doesn't take into account is that the file input stream does prefetching. that prefetching consumes memory. I feel like we need to have a bounded set of readers or a shared pool of them or smth.